### PR TITLE
enhancement: allow disabling download csv on a shared view

### PR DESCRIPTION
**Pull request by wingkwong**
## Change Summary

- Add a boolean column `meta` in `nc_views_v2`. By default, `meta. allowCSVDownload` is `true` in order to cover the existing cases.
- Add a new option for users to choose if there is a download csv option in shared view (ref: #807)
- Render MoreActions based on the flag - Currently the only option in MoreActions is download csv
- Include Download Allowed column in Shared View List. Show '✔️ or ❌ for grid view only. N/A otherwise.
- Update related cypress tests

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [x] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Enable Download (Default)

image

image

Disable Download

image

image

Shared View List

image

### CODE REVIEW

File: packages/nocodb/src/lib/migrations/v2/nc_018_add_download_in_view.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import Knex from 'knex';
+import { MetaTable } from '../../utils/globals';
+
+const up = async (knex: Knex) => {
+  await knex.schema.alterTable(MetaTable.VIEWS, (table) => {
+    table.boolean('download').defaultTo(true);
```
Comment (pranavxc) [inline packages/nocodb/src/lib/migrations/v2/nc_018_add_download_in_view.ts:6]: What about keeping a meta(`text` type) column that holds a JSON string?  Extending view-specific metadata becomes more manageable that way we don't have to write more migrations.

**Review by pranavxc — APPROVED** [review]

**Comment by dstala** [comment]
@wingkwong 
Locally also I could see behavior where on first click `share view`, modal did not open. I am assuming it to be with browser cache or some state maintained in the browser (just a guess).  

I have captured responses for the first click & subsequent click here
In response, meta was set to null in first & was set to `meta: "{\"allowCSVDownload\":true}"` in next

```
{"id":"vw_pgm7npenhzh557","base_id":"ds_n0tt53v2vtm3s3","project_id":"p_qyrmkywgv9qo0t","fk_model_id":"md_iclrlpb3hb406d","title":"Payment","type":3,"is_default":1,"show_system_fields":null,"lock_type":"collaborative","uuid":"6f2b9206-2858-460b-a611-4032b68fb399","password":null,"show":1,"order":1,"created_at":"2022-07-09 02:54:30","updated_at":"2022-07-09 02:54:30","meta":null}
```

```
{"id":"vw_pgm7npenhzh557","base_id":"ds_n0tt53v2vtm3s3","project_id":"p_qyrmkywgv9qo0t","fk_model_id":"md_iclrlpb3hb406d","title":"Payment","type":3,"is_default":1,"show_system_fields":null,"lock_type":"collaborative","uuid":"6f2b9206-2858-460b-a611-4032b68fb399","password":null,"show":1,"order":1,"created_at":"2022-07-09 02:54:30","updated_at":"2022-07-09 02:54:30","meta":"{\"allowCSVDownload\":true}"}
```

**Comment by wingkwong** [comment]
@dstala Yes. Currently the behaviour is a bit changed. The options will be stored in `meta` and will be persisted, i.e. when the switcher is off, next time when you open the share view dialog, it's still off as it's reading the value from `meta`.

**Review by mertmit — COMMENTED** [review]
In my opinion handling view meta's parse and stringify operations within view model is more appropriate(as in column meta). This way we can assume meta is an object (no need to parse or stringify every time)
Waiting your opinion on this @wingkwong

**Review by mertmit — APPROVED** [review]

**Pull request merged**